### PR TITLE
deploy: drop custom-server-block and use status_listen

### PR DESCRIPTION
**Pull request by hbagdi**
Until Kong 1.4, there didn't exist a way to health-check kong and expose
metrics without also exposing the admin api of kong.
In kong 1.4, a new server KONG_STATUS_LISTEN was added which provides
these capabilities.

This commit removes the reliance on the custom server block that was
provided via a ConfigMap.
With 0.9, the next version of controller, the minimum version of kong
should be 1.4, hence this breaking change.

The error logs have been modified as well to include proxy error logs as
well.

### CODE REVIEW

**Comment by hbagdi** [comment]
This should be merged after #633

File: deploy/manifests/base/kong-ingress-dbless.yaml [hunk 1]
```diff
@@ -28,20 +28,28 @@ spec:
...
         - name: KONG_ADMIN_ACCESS_LOG
           value: /dev/stdout
         - name: KONG_ADMIN_ERROR_LOG
           value: /dev/stderr
-        - name: KONG_ADMIN_LISTEN
-          value: 127.0.0.1:8444 ssl
-        - name: KONG_PROXY_LISTEN
-          value: 0.0.0.0:8000, 0.0.0.0:8443 ssl http2
+        # - name: KONG_PROXY_ACCESS_LOG
+        # - value: /dev/stdout
```
Comment (rainest) [inline deploy/manifests/base/kong-ingress-dbless.yaml:50]: Was commenting these intentional? It looks like this is unrelated.

File: deploy/manifests/base/kong-ingress-dbless.yaml [hunk 2]
```diff
@@ -54,12 +62,12 @@ spec:
           containerPort: 8443
           protocol: TCP
         - name: metrics
-          containerPort: 9542
+          containerPort: 8100
           protocol: TCP
         livenessProbe:
           httpGet:
             path: /health
```
Comment (rainest) [inline deploy/manifests/base/kong-ingress-dbless.yaml:69]: This should now be `/status`, correct, or is there a `/health` endpoint in newer versions? In Enterprise 1.5:

```
# curl -s localhost:8100/health
{"message":"Not found"}

# curl -s localhost:8100/status; echo
{"database":{"reachable":true}, ... "connections_waiting":0}}
```
Applies to the other `/health` checks for readiness and other manifests also. Have not marked them all because they're easy enough to find with grep and the review comments would get rather noisy with each.

File: deploy/manifests/base/kong-ingress-dbless.yaml (hunk 1 above)
Comment (hbagdi) [reply deploy/manifests/base/kong-ingress-dbless.yaml:50]: Yes, this is intentional.
Users when they are getting started can use this to log proxy access logs quickly.

File: deploy/manifests/base/kong-ingress-dbless.yaml (hunk 2 above)
Comment (hbagdi) [reply deploy/manifests/base/kong-ingress-dbless.yaml]: My bad. I fixed and re-ran the build script.
